rclrs/src/rcl_utils.rs

```diff
@@ -0,0 +1,101 @@
+use crate::rcl_bindings::*;
+use crate::{RclrsError, ToResult};
+
+use libc::c_void;
+use std::ptr::null_mut;
+
+/// Trait used for obtaining information from rcl_arguments_t.
+pub(crate) trait RclGetArg {
+    /// Get count function from rcl library (e.g. rcl_arguments_get_count_unparsed)
+    unsafe fn get_count(rcl_args: *const rcl_arguments_t) -> ::std::os::raw::c_int;
+
+    /// Get function from rcl library (e.g. rcl_arguments_get_unparsed)
+    unsafe fn get_indices(
+        rcl_args: *const rcl_arguments_t,
+        allocator: rcl_allocator_t,
+        out_ptr: *mut *mut ::std::os::raw::c_int,
+    ) -> rcl_ret_t;
+}
+
+/// Get ROS unparsed arguments tag.
+///
+/// Tag designated to be used as generic argument of [`get_rcl_arguments`].
+pub(crate) struct UnparsedRos;
+
+/// Get non-ROS unparsed arguments tag.
+///
+/// Tag designated to be used as generic argument of [`get_rcl_arguments`].
+pub(crate) struct UnparsedNonRos;
+
+/// Macro to avoid boilerplate code for above tags.
+///
+/// tag - one of above tags
+/// get_count_function - function for getting count of given arguments type
+/// get_indices_function - function for getting indices of given arguments type
+macro_rules! impl_get_arg {
```

Comment:
Is the macro here really necessary? Will we be using this for more cases than just the above two tags?

Additionally, it creates two `unsafe` functions that won't have (and possibly can't have) safety comments, which makes me a little nervous.

@nnmm, @esteve, what are your thoughts?

Reply:
Well, I've only been focusing on the high-level design so far, see https://github.com/ros2-rust/ros2_rust/pull/210#issuecomment-1172905936 and https://github.com/ros2-rust/ros2_rust/pull/210#issuecomment-1175018118. Until that's settled maybe it's a bit early to concern ourselves with implementation details.

Reply:
Doesn't seem to me that it'll be used much more than that, I'd rewrite this as a function

Reply:
@nnmm I don't think it's never too early, it's fair to provide any feedback while we review this PR

Reply:
So it's ok to leave this rcl_utils.rs file and rewrite only that macro part, is it?